src/cycles/basis.jl

```diff
@@ -0,0 +1,72 @@
+# Code in this file inspired by NetworkX.
+
+"""
+    cycle_basis(g, root=nothing)
+
+Returns a list of cycles which form a basis for cycles of `g`.
+
+A basis for cycles of a network is a minimal collection of
+cycles such that any cycle in the network can be written
+as a sum of cycles in the basis.  Here summation of cycles
+is defined as "exclusive or" of the edges. Cycle bases are
+useful, e.g. when deriving equations for electric circuits
+using Kirchhoff's Laws.
+
+# Arguments
```

Comment:
This isn't necessary. Include the argument explanation in the description (see above).